ipod-shuffle-4g.py

```diff
@@ -322,14 +323,20 @@ def construct(self):
             track = Track(self)
             verboseprint("[*] Adding track", i)
             track.populate(i)
+            if track.is_podcast:
+                self.total_podcasts += 1
```

Comment:
If you are counting here, shouldn't you first reset the value to 0?

Also why is `["number_of_tracks"]` an array and `total_podcasts` a direct property? (I dont know the exact python terms, but I think you understand the difference)

Reply:
Maybe unknown1 is the value of the podcast count? Do you have an itunes to test that?

Reply:
I'm not sure what you mean by "value of the podcast count". The number of podcast tracks in a given playlist?
Either way, I wouldn't expect a `long long` to be needed for that value. Also, if the people who reverse engineered and documented the database format haven't figured out those "unknown" sections, I seriously doubt I would.

Apple also keeps deprecating iPod support and diminishing where it can accessed. For iPod Shuffle that's currently the Finder and it's a bit buggy.

Oh! The format written out by that support is also different entirely. The version string is different and several fields take up more bytes. Documenting it would be a big effort to determine the updated format. I'm actually a bit interested in that, but not any time soon.

Reply:
> If you are counting here, shouldn't you first reset the value to 0?

The value already starts as zero for each `TrackHeader` as defined in the `_struct`. It's incremented and the written out to the database and isn't reused.

> Also why is `["number_of_tracks"]` an array and `total_podcasts` a direct property? (I dont know the exact python terms, but I think you understand the difference)

That's because `["number_of_tracks"]` is defined in the `_struct` and using the `[""]` notation accesses the value via the `__getitem__` method defined in the `Record` object.
`total_podcasts` is defined directly in the `TrackHeader` object, right below `base_offset`.

Reply:
The guy that reverse engineered this was mostly me. There was a basis, but it was missing a lot. But I remember that I have never added podcasts, so this knowledge is not from me, but we could find out, if we check the hex. But if it still works on the ipod, I dont see any major issues.

Reply:
Oh cool, well, hats off and thanks for your work.

That's a fair point about calling it twice, though the script doesn't currently do that. I'll make the change.

It does indeed work as is, so I'd be hesitant to make extra changes.